pkg/backup/abs/abs.go

```diff
@@ -89,15 +90,23 @@ func (w *ABS) Put(key string, r io.Reader) error {
 	blobName := path.Join(v1, w.prefix, key)
 	blob := w.container.GetBlobReference(blobName)
 
+	var buf bytes.Buffer
+	if r != nil {
+		_, err := io.Copy(&buf, r)
+		if err != nil {
+			return err
+		}
+		r = &buf
+	}
+
 	err := blob.CreateBlockBlob(&storage.PutBlobOptions{})
 	if err != nil {
 		return err
 	}
 
-	buf := new(bytes.Buffer)
-	buf.ReadFrom(r)
```

Comment:
This is the reason we get partial bytes from io.Reader. We forget to check the error of this function. 
The fix is above.

Reply:
Why do we need to copy to find the error ?